index.html

```diff
@@ -207,36 +208,81 @@ <h2>Legend</h2>
        return p;
      }
 
-     function sanitizeHTML(value) {
-       // Only support links and code.
-       const template = document.createElement('template');
-       const div = template.content.ownerDocument.createElement('div');
-       template.content.append(div); // force parser context to "in body"
-       div.innerHTML = value;
-       const iterator = document.createNodeIterator(div);
-       let node = iterator.nextNode(); // This is the div
-       while (node = iterator.nextNode()) {
-         if (node.nodeType === Node.TEXT_NODE) {
-           continue;
+     function expandInline(markdown) {
+       const fragment = document.createDocumentFragment();
+       const codePattern = /`([^`]+)`/;
+
+       let cursor = 0;
+
+       while (true) {
+         const match = codePattern.exec(markdown);
+         if (!match) {
+           // No more code segments, just append remaining text (if any)
+           if (markdown) {
+             fragment.appendChild(document.createTextNode(markdown));
+           }
+           break;
          }
-         if (node.nodeType === Node.ELEMENT_NODE) {
-           if (node instanceof HTMLAnchorElement) {
-             if (node.attributes.length === 1 && node.hasAttribute('href')) {
-               const href = node.getAttribute('href');
-               if (href[0] === '#' || href.startsWith("https://")) {
-                 continue;
-               }
-             }
-           } else if (node instanceof HTMLElement && node.localName === "code") {
-             if (node.attributes.length === 0) {
-               continue;
-             }
+
+         const textBefore = markdown.slice(cursor, match.index);
+         if (textBefore) {
+           fragment.appendChild(document.createTextNode(textBefore));
+         }
+
+         // Create a <code> element for the matched segment
+         const codeElement = document.createElement('code');
+         codeElement.textContent = match[1];
+         fragment.appendChild(codeElement);
+
+         // Move our cursor beyond this match
+         cursor = match.index + match[0].length;
+         // Slice off what we've already processed
+         markdown = markdown.slice(cursor);
+         // Reset cursor for next iteration
+         cursor = 0;
+       }
+
+       return fragment;
+     }
+
+     function expandLinks(markdown) {
+       const fragment = document.createDocumentFragment();
+       const linkPattern = /\[([^\]]+)\]\(((?:#|https:\/\/)[^\)]+)\)/;
```

Comment:
Do you want to match on bare links as well?  Like `/\b(https:\/\/\S+)\b/`?  (Save that for later perhaps.)

Reply:
No, the data should have links marked up.